cpp/src/text/tokenize.cu

```diff
@@ -150,6 +151,68 @@ std::unique_ptr<cudf::column> count_tokens(cudf::strings_column_view const& stri
     stream);
 }
 
+// tokenize on every character
+std::unique_ptr<cudf::column> character_tokenize(cudf::strings_column_view const& strings_column,
+                                                 cudaStream_t stream,
+                                                 rmm::mr::device_memory_resource* mr)
+{
+  auto strings_count = strings_column.size();
+  if (strings_count == 0) { return cudf::make_empty_column(cudf::data_type{cudf::STRING}); }
+
+  auto execpol   = rmm::exec_policy(stream);
+  auto d_offsets = strings_column.offsets().data<int32_t>();
+  d_offsets += strings_column.offset();  // nvbug-2808421 : do not combine with the previous line
+  auto offset      = thrust::device_pointer_cast(d_offsets)[0];
```

Comment:
This is problematic because it can't be stream ordered. I'd suggest making a new detail API similar to `get_element`, but returns a primitive type instead of a `cudf::scalar`. E.g.,
```
template <typename T>
T get_value(column_view const& c, size_type element_index, cudaStream_t s)
```

It has the added benefit of being easier to read and understand what's going on.

Reply:
Added a `cudf::detail::get_value` API to do wrap a cudaMemcpy that is stream aware.